runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -57,23 +73,40 @@
 @Slf4j
 @PluginDescriptor(
 	name = "Mining",
-	description = "Show ore respawn timers",
-	tags = {"overlay", "skilling", "timers"},
+	description = "Show ore respawn timers and coal bag overlay",
+	tags = {"overlay", "skilling", "timers", "coal", "coalbag", "coal bag"},
 	enabledByDefault = false
 )
 public class MiningPlugin extends Plugin
 {
 	private static final int ROCK_DISTANCE = 14;
 	private static final int MINING_GUILD_REGION = 12183;
 
+	private static final Pattern COAL_BAG_EMPTY_MESSAGE = Pattern.compile("^The coal bag is (now )?empty\\.$");
+	private static final Pattern COAL_BAG_ONE_MESSAGE = Pattern.compile("^The coal bag contains one piece of coal\\.$");
+	private static final Pattern COAL_BAG_AMOUNT_MESSAGE = Pattern.compile("^The coal bag contains (\\d+) pieces of coal\\.$");
+
+	private static final int MAX_INVENTORY_SPACE = 28;
+	private static final int FULL_COAL_BAG_AMOUNT = 27;
```

Comment:
When wearing a smithing cape you can store up to 36 coal in the bag.

Reply:
Whats the status on this? Is it going to be fixed?